src/components/CreateCourse/CreateCourse.jsx

```diff
@@ -0,0 +1,211 @@
+import { useState } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import Input from '../Input/Input';
+import Button from '../Button/Button';
+import Message from '../Message/Message';
+import formatDuration from '../../mixins/format-duration';
+
+const CreateCourse = (props) => {
+	const [allAuthors, setAllAuthors] = useState(props.authors);
+	const [authorslist, setAuthorsList] = useState(props.authors);
+	const [newAuthorName, setNewAuthorName] = useState('');
+	const [courseAuthors, setCourseAuthors] = useState([]);
+	const [title, setTitle] = useState('');
+	const [description, setDescription] = useState('');
+	const [duration, setDuration] = useState('');
+	const [userDuration, setUserDuration] = useState('00:00 hours');
+
+	const addNewAuthor = (name) => {
+		if (!name || name.trim() === '' || name.length < 2) {
+			alert('Please, enter valid name!');
+			return false;
+		}
+
+		let newAuthor = {
+			id: uuidv4(),
+			name: name,
+		};
+
+		setAuthorsList((oldAuthorslist) => [...oldAuthorslist, newAuthor]);
+		setAllAuthors((oldAuthors) => [...oldAuthors, newAuthor]);
+		setNewAuthorName('');
+		console.log(allAuthors);
+	};
+
+	const addNewAuthorToCourse = (author) => {
+		setCourseAuthors((oldCourseAuthorslist) => [
+			...oldCourseAuthorslist,
+			author,
+		]);
+		setAuthorsList(authorslist.filter((el) => el.id !== author.id));
+	};
+
+	const deleteCourseAuthor = (author) => {
+		setCourseAuthors(courseAuthors.filter((el) => el.id !== author.id));
+		setAuthorsList((oldAuthorslist) => [...oldAuthorslist, author]);
+	};
+
+	const getUserDuration = (minutes) => {
+		setDuration(minutes);
+		setUserDuration(formatDuration(minutes));
+	};
+
+	const validateForm = () => {
+		if (!title || title.trim() === '') {
+			alert('Please, fill in all fields!');
+			return false;
+		}
+		if (!description || description.trim() === '' || description.length < 2) {
+			alert('Please, fill in all fields!');
+			return false;
+		}
+		if (!duration || duration <= 0) {
+			alert('Please, enter duration of the course in minutes!');
+			return false;
+		}
+		return true;
+	};
+
+	const createNewCourse = (e) => {
+		e.preventDefault();
+
+		let authorIdList = [];
+		courseAuthors.map((author) => {
+			return authorIdList.push(author.id);
+		});
```

Comment:
Need to refactor `const authorIdList = courseAuthors.reduce((acc, curr) => {....}, [])`